dist/index.js

```diff
@@ -287,38 +275,47 @@ class Utils {
     static computeCacheKey(appendedCacheKey) {
         return __awaiter(this, void 0, void 0, function* () {
             let key = "";
-            const inputVcpkgPath = core.getInput(runvcpkglib.vcpkgDirectory);
+            const restoreKeys = [];
             const actionLib = new actionlib.ActionLib();
             const baseUtil = new baseutillib.BaseUtilLib(actionLib);
+            key += "runnerOS=" + process.env.ImageOS ? process.env.ImageOS : 0;
+            // Add the triplet only if it is provided.
+            const triplet = core.getInput(runvcpkglib.vcpkgTriplet);
+            if (triplet) {
+                key += "-triplet=" + triplet;
+            }
+            restoreKeys.push(key);
+            if (process.env.ImageVersion) {
+                key += "_ghActionsEnvironmentVersion=" + process.env.ImageVersion;
+            }
+            restoreKeys.push(key);
+            const inputVcpkgPath = core.getInput(runvcpkglib.vcpkgDirectory);
             const [commitId, isSubmodule] = yield Utils.getVcpkgCommitId(baseUtil, inputVcpkgPath);
             const userProvidedCommitId = core.getInput(runvcpkglib.vcpkgCommitId);
             if (commitId) {
                 core.info(`vcpkg identified at commitId='${commitId}', adding it to the cache's key.`);
                 if (isSubmodule) {
-                    key += `submodGitId=${commitId}`;
+                    key += `_vcpkgGitCommit=${commitId}`;
                 }
                 else {
-                    key += "localGitId=" + Utils.hashCode(userProvidedCommitId);
+                    key += `_vcpkgGitCommit=${userProvidedCommitId}`;
                 }
+                restoreKeys.push(key);
             }
             else if (userProvidedCommitId) {
                 core.info(`Using user provided vcpkg's Git commit id='${userProvidedCommitId}', adding it to the cache's key.`);
-                key += "localGitId=" + Utils.hashCode(userProvidedCommitId);
+                key += `_vcpkgGitCommit=${userProvidedCommitId}`;
+                restoreKeys.push(key);
             }
             else {
                 core.info(`No vcpkg's commit id was provided, does not contribute to the cache's key.`);
             }
```

Comment:
I would like to put the hash of the vcpkg.json file before the commit hash (#95) in the cache key. Once a project is setup I would think the vcpkg.json file would change infrequently. Updating the packages would simply be a matter of changing the commit hash/updating the submodule. When doing so, the cache should hit for the last cache with the same vcpkg.json to avoid having to rebuild every package. But this needs to wait for the `hashFiles` function to be published in the @actions/glob NPM package (https://github.com/actions/toolkit/pull/830).

Reply:
is the `hashFiles` function really necessary? Or any hash function should be a good fit?

Reply:
I suppose another hash function could do :shrug:

Reply:
I still see that your point of using eactly the same hash function has its benefits: it would not invalidate existing keys. On the other hands, this PR already invalidate existing keys as the key changed its format, so no concern here. Any hash function would be a good fit.